### DISCUSSION TITLE
How to install pyscript-related modules (e.g. pyodide, pyweb, js) locally for local development

### DISCUSSION THREAD

**Discussion by cosmicBboy** [pyscript/pyscript#2080] [Q&A]
Hi pyscript devs!

Loving pyscript so far, is there a way to install pyscript-related packages so that I can see function definitions/autocompletes in my local vscode IDE?

<img width="617" alt="image" src="https://github.com/pyscript/pyscript/assets/2816689/9f95124c-5bac-4072-b1f5-6e0bd520444e">

Currently they are unresolved. Right now I'm using the docs / looking at source code to know what functions are available and what their signatures are. Would be cool to be able to see what these are in the IDE.

**Comment by WebReflection** [comment]
when it comes to `js` it's literally the entirety of the *JavaScript* global context, it's not a module we provide, it's the *FFI* that bridges Python to JS world (and vice-versa). I don't know if you can tell your IDE to consider that a `wndow` or a `globalThis` in the *JS* world, as I don't think IDEs are capable to merge two distinct worlds within the same code, plus these don't have memory and references ability to map everything correctly so this is a hard one to me, although I don't know all IDEs and I don't know if you cna force a foreign type there to make your IDE happy.
